contracts/eosio.bios/src/eosio.bios.cpp

```diff
@@ -17,6 +18,34 @@ void bios::setabi( name account, const std::vector<char>& abi ) {
    }
 }
 
+void bios::setfinalizer( const finalizer_policy& finalizer_policy ) {
+   require_auth( get_self() );
+   check(finalizer_policy.finalizers.size() > 0, "require at least one finalizer");
+
+   eosio::abi_finalizer_policy abi_finalizer_policy;
+   abi_finalizer_policy.fthreshold = finalizer_policy.threshold;
+   abi_finalizer_policy.finalizers.reserve(finalizer_policy.finalizers.size());
+
+   const std::string pk_prefix = "PUB_BLS";
+   const std::string sig_prefix = "SIG_BLS";
+
+   for (const auto& f: finalizer_policy.finalizers) {
+      // basic key format checks
+      check(f.public_key.substr(0, pk_prefix.length()) == pk_prefix, "public key not started with PUB_BLS");
+      check(f.pop.substr(0, sig_prefix.length()) == sig_prefix, "proof of possession signature not started with SIG_BLS");
+
+      // proof of possession of private key check
+      const auto pk = eosio::decode_bls_public_key_to_g1(f.public_key);
+      const auto signature = eosio::decode_bls_signature_to_g2(f.pop);
+      check(eosio::bls_pop_verify(pk, signature), "proof of possession failed");
+
+      const std::vector<char> pk_vector(pk.begin(), pk.end());
+      abi_finalizer_policy.finalizers.emplace_back(eosio::abi_finalizer_authority{f.description, f.weight, pk_vector});
```

Comment:
Make `pk_vector` non-const and then `std::move(pk_vector)` here.
I think you could take `finalizer_policy ` by value and move `f.description` as well.